### ISSUE TITLE
HSQLDB cannot handle negative DAY intervals when using expression syntax

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10256]
We're currently translating `currentDate().plus(1)` as follows in HSQLDB:

```sql
-- Input
current_date + (-1)

-- Output
current_date + -1 day
```

This doesn't work in HSQLDB, which doesn't allow for expressions to precede the `DAY` keyword, unless they're in parentheses. The error is:

```
Caused by: org.hsqldb.HsqlException: incompatible data type in operation
	at org.hsqldb.error.Error.error(Unknown Source)
	at org.hsqldb.error.Error.error(Unknown Source)
	at org.hsqldb.ExpressionArithmetic.resolveTypes(Unknown Source)
	at org.hsqldb.ExpressionArithmetic.resolveTypes(Unknown Source)
	at org.hsqldb.QuerySpecification.resolveExpressionTypes(Unknown Source)
	at org.hsqldb.QuerySpecification.resolveTypesPartOne(Unknown Source)
	at org.hsqldb.QueryExpression.resolve(Unknown Source)
	at org.hsqldb.ParserDQL.compileCursorSpecification(Unknown Source)
	at org.hsqldb.ParserCommand.compilePart(Unknown Source)
	at org.hsqldb.ParserCommand.compileStatements(Unknown Source)
	at org.hsqldb.Session.executeDirectStatement(Unknown Source)
	at org.hsqldb.Session.execute(Unknown Source)
	... 48 more
```

This works:

```sql
-- Output
current_date + (-1) day
```

Given that expressions could be used in `Field.plus(Field)`, we should probably always generate these parentheses.

**Comment by lukaseder** [comment]
The fix will also affect the DB2 implementation

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.3 (#10257)
